chain/bitcoind_client.go

```diff
@@ -1094,15 +1041,13 @@ func (c *BitcoindClient) filterBlock(block *wire.MsgBlock, height int32,
 	// Update the expiration map by setting the block's confirmed
 	// transactions and deleting any in the mempool that were confirmed
 	// over 288 blocks ago.
-	c.clientMtx.Lock()
```

Comment:
why safe to remove mutex?

Reply:
Mempool checks are now shared with `watchMtx`. It's documented in the struct definition.

Reply:
Fixed.